OpenRA.Mods.Common/Traits/Buildings/BuildingUtils.cs

```diff
@@ -112,9 +112,17 @@ public static IEnumerable<(CPos Cell, Actor Actor)> GetLineBuildCells(World worl
 					if (dirs[d] != 0)
 						continue;
 
+					var si = ai;
+					var sbi = bi;
```

Comment:
While we're here: Can we use `segmentInfo` and `segmentBuildingInfo`? (You don't need to change `ai` and `bi`, but we don't have to introduce more variable with bad names.)